### PR TITLE
KHWE-27:  add songs to my songs

### PR DESCRIPTION
**Pull request by lubovgribiniyk**

### CODE REVIEW

File: src/components/DotsMenu/DotsMenu.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ export class DotsMenuComponent {
     const activeMenuItem = e.target.closest(".dots-menu__item");
     if (activeMenuItem) {
       const id = parseInt(activeMenuItem.dataset.id, 10);
-      this.props.items[id].handler();
+      this.props.items[id].handler(e);
```
Comment (kozak-iz-kh) [inline src/components/DotsMenu/DotsMenu.js]: do not pass e - use bind and partial application

File: src/components/MediaPlayer/MediaPlayer.js [hunk 2]
```diff
@@ -73,6 +75,13 @@ export class MediaPlayerComponent {
     this.props.onDialogOpen();
   }
 
+  handleAddSong() {
+    MusicService.setUserSong(
+      AuthService.getCurrentUser().uid,
+      this.song.id
+    ).then(() => this.props.onAddSong());
```
Comment (kozak-iz-kh) [inline src/components/MediaPlayer/MediaPlayer.js]: then(this.props.onAddSong)

File: src/components/MySongs/MySongs.js [hunk 3]
```diff
@@ -54,11 +56,27 @@ export class MySongsComponent {
...
   }
 
+  handleRemoveSong(songId) {
+    MusicService.removeUserSong(AuthService.getCurrentUser().uid, songId).then(
+      () => this.mount()
+    );
+  }
+
+  addSong() {
+    if (this.mountPoint.querySelector(".my-songs")) {
```
Comment (kozak-iz-kh) [inline src/components/MySongs/MySongs.js:67]: move to queryselectors

File: src/components/MySongs/MySongs.js [hunk 4]
```diff
@@ -54,11 +56,27 @@ export class MySongsComponent {
...
+      this.mount();
+    }
+  }
+
   mountChildren() {
+    if (!this.songs.length) {
```
Comment (kozak-iz-kh) [inline src/components/MySongs/MySongs.js]: is it possible to move this check to mount?

File: src/components/MySongs/MySongs.js (hunk 3 above)
Comment (lubovgribiniyk) [reply src/components/MySongs/MySongs.js:67]: This way doesn't work in the case when this component is mount at least once

File: src/components/MySongs/MySongs.js [hunk 5]
```diff
@@ -1,12 +1,14 @@
 import mySongs from "./MySongs.html";
```
Comment (ihor4568) [inline src/components/MySongs/MySongs.js]: mySongsTemplate

File: src/components/MySongs/MySongs.html [hunk 6]
```diff
@@ -1,4 +1,13 @@
 <div class="my-songs">
+  <% if (isDataFetched) { %> <% if (hasSongs) { %>
```
Comment (ihor4568) [inline src/components/MySongs/MySongs.html:2]: please format these tags correctly - move new `if` and new `}` to the new lines

File: src/components/Main/Main.js [hunk 7]
```diff
@@ -229,7 +233,8 @@ export class MainComponent {
     this.player = new MediaPlayerComponent(this.playerPoint, {
       onDialogOpen: this.handleDialogOpen.bind(this),
       onPlayerChangeState: this.handlePlayerChangeState.bind(this),
-      onLegalOptionClick: this.handleSetInfo.bind(this)
+      onLegalOptionClick: this.handleSetInfo.bind(this),
+      onAddSong: this.handleAddSong.bind(this)
```
Comment (ihor4568) [inline src/components/Main/Main.js]: there is no need for this handleAddSong. You can just directly this.table.addSong.bind(this.table)

**Review by OlegShynkarenko — APPROVED** [review]

**Pull request merged**